### DISCUSSION TITLE
실습을 위한 Gazebo 환경과 사용법 소개 ERROR

### DISCUSSION THREAD

**Discussion by Bruceworld0** [Road-Balance/gcamp_ros_basic#7] [Q&A]
thank you for a great course

3. 실습을 위한 Gazebo 환경과 사용법 소개
$ git clone https://github.com/Road-Balance/gcamp_ros_basic.git
$ cd ~/gcamp_ws
$ catkin_make

and error like below...

how can i solve the problem?

/home/b/gcamp_ws/src/gcamp_ros_basic/cpp_service_tutorial/src/robot_turning_srv.cpp:10:10: fatal error: cpp_service_tutorial/ControlTurningMessage.h: No such file or directory
 #include "cpp_service_tutorial/ControlTurningMessage.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_server.dir/build.make:62: recipe for target 'gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_server.dir/src/robot_turning_srv.cpp.o' failed
make[2]: *** [gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_server.dir/src/robot_turning_srv.cpp.o] Error 1
CMakeFiles/Makefile2:1912: recipe for target 'gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_server.dir/all' failed
make[1]: *** [gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_server.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 31%] Generating Javascript code from action_tutorial/MazeAction.msg
/home/b/gcamp_ws/src/gcamp_ros_basic/cpp_service_tutorial/src/robot_turning_client.cpp:8:10: fatal error: cpp_service_tutorial/ControlTurningMessage.h: No such file or directory
 #include "cpp_service_tutorial/ControlTurningMessage.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_client.dir/build.make:62: recipe for target 'gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_client.dir/src/robot_turning_client.cpp.o' failed
make[2]: *** [gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_client.dir/src/robot_turning_client.cpp.o] Error 1
CMakeFiles/Makefile2:1763: recipe for target 'gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_client.dir/all' failed
make[1]: *** [gcamp_ros_basic/cpp_service_tutorial/CMakeFiles/robot_turning_client.dir/all] Error 2
[ 33%] Generating Javascript code from action_tutorial/MazeResult.msg
[ 34%] Generating C++ code from action_tutorial/MazeActionResult.msg
[ 36%] Generating Javascript code from action_tutorial/MazeGoal.msg
[ 37%] Generating C++ code from action_tutorial/MazeActionFeedback.msg
[ 39%] Generating C++ code from action_tutorial/MazeFeedback.msg
[ 39%] Built target action_tutorial_generate_messages_nodejs
[ 40%] Generating C++ code from action_tutorial/MazeAction.msg
[ 40%] Built target service_tutorial_generate_messages_eus
[ 42%] Generating C++ code from action_tutorial/MazeResult.msg
[ 43%] Generating C++ code from action_tutorial/MazeGoal.msg
[ 43%] Built target action_tutorial_generate_messages_cpp
[ 44%] Linking CXX executable /home/b/gcamp_ws/devel/lib/cpp_topic_tutorial/basic_node
[ 46%] Linking CXX executable /home/b/gcamp_ws/devel/lib/cpp_topic_tutorial/cmd_vel_pub_node
[ 47%] Linking CXX executable /home/b/gcamp_ws/devel/lib/cpp_service_tutorial/spawn_model_client
[ 47%] Built target cmd_vel_pub_node
[ 47%] Built target basic_node
[ 47%] Built target spawn_model_client
[ 49%] Linking CXX executable /home/b/gcamp_ws/devel/lib/cpp_topic_tutorial/laser_sub_node
[ 50%] Linking CXX executable /home/b/gcamp_ws/devel/lib/cpp_topic_tutorial/parking_node
[ 50%] Built target laser_sub_node
[ 50%] Built target parking_node
[ 52%] Linking CXX executable /home/b/gcamp_ws/devel/lib/cpp_action_tutorial/fibonacci_action_server
[ 52%] Built target fibonacci_action_server
[ 53%] Linking CXX executable /home/b/gcamp_ws/devel/lib/cpp_action_tutorial/fibonacci_action_client
[ 53%] Built target fibonacci_action_client
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
Invoking "make -j12 -l12" failed

**Comment by kimsooyoung** [comment]
**cpp_service_tutorial** contains custom msg type and you must build that first before `cma`
Try to type below command lines.

```
$ cd ~/gcamp_ws
$ cop cpp_service_tutorial  
$ cop cpp_topic_tutorial
```

That's known issue and I'll gonna add comment in Notion Pages right away.
Plz leave feedback after that.

Thank you

**Reply by Bruceworld0** [reply]
It works!! Thank you!
